website/cue/reference/remap/functions/type_def.cue

```diff
@@ -0,0 +1,31 @@
+package metadata
+
+remap: functions: type_def: {
+	category: "Debug"
+	description: """
+		A function that returns a representation of the internal compile-time type definition of a variable. This function
+		is intended for debugging / development only and should not be used in production. This function is not considered stable
+		and can change at any time.
+		"""
```

Comment:
Should we simply _not_ document this function? We have other function(s) we don't document, so it wouldn't be the first time we make that decision.

Reply:
I'd personally like to keep this internal, at least until we're a bit further in VRL's development. I realize the docs say it's unstable/internal/etc, but I still worry that we expose this, users start relying on it, and we have to go through a bunch of hoops to warn people when things change w.r.t. our type system in a way that breaks the output of this function.

I know the chances of that happening are slim, but they are still there.

My preference would be:

- move this function out of `stdlib`
- move it into the `compiler` crate
- have it behind a feature gate to enable/disable it
- disable it when we build releases, but keep it in otherwise

As for bug reports: as long as they provide (or we ask for) the program causing the issue, we can fairly trivially run the program ourselves with this function enabled, and injected.

Reply:
I should also say I'm happy to take on the extra work, as I understand it adds complexity to a fairly trivial PR that you wanted to squeeze in between the other work. I'm already working on VRL diagnostic/compiler related improvements, so I can slot one in as well.

Reply:
I'm fine with moving it behind as feature flag if that flag is enabled by default for local builds. (Which I think is what you are saying). Feel free to move this yourself. I have other things I need to do now, but if you don't get to it soon I might come back to this.

Reply:
I looked into this, but it wasn't as easy to enable this locally by default, but have it be removed in release builds. Instead, I opted to remove the public docs, and add a note to the implementation that this is supposed to be used internally only.

Let's hope this is enough to discourage people from using it in production. 🤞